ocaml/xapi-storage-cli/main.ml

```diff
@@ -300,10 +303,13 @@ let on_vdi f common_opts sr vdi =
       `Error (true, "must supply VDI")
   | Some sr, Some vdi ->
       let sr, vdi = Storage_interface.(Sr.of_string sr, Vdi.of_string vdi) in
-      wrap common_opts (fun () -> f sr vdi)
+      wrap_exn common_opts (fun () -> f sr vdi)
+
+let on_vdi' f common_opts sr vdi =
```

Comment:
I think having `on_vdi` and `on_vdi'` in scope is dangerous if they have the same type. I might be better to shadow `on_vdi` in that case.